src/components/HeaderForm.js

```diff
@@ -300,23 +342,35 @@ class HeaderForm extends Component {
     }
   };
   getNextViewTarget = () => ({
+    tracks: new Array(create_track({name: this.state.graphFile, type: fileTypes.GRAPH}), 
+                      create_track({name: this.state.gbwtFile, type: fileTypes.HAPLOTYPE}), 
+                      create_track({name: this.state.gamFile, type: fileTypes.READ}), 
+                      create_track({name: this.state.bedFile, type: fileTypes.BED})),
     name: this.state.name,
     region: this.state.region,
-    graphFile: this.state.graphFile,
-    gbwtFile: this.state.gbwtFile,
-    gamFile: this.state.gamFile,
-    bedFile: this.state.bedFile,
     dataPath: this.state.dataPath,
     dataType: this.state.dataType,
   });
 
   handleGoButton = () => {
+    console.log("HANDLING GO BUTTON:");
     if (this.props.dataOrigin !== dataOriginTypes.API) {
       this.props.setColorSetting("haplotypeColors", "ygreys");
       this.props.setColorSetting("forwardReadColors", "reds");
     }
-    const viewTarget = this.getNextViewTarget();
-    this.props.setCurrentViewTarget(viewTarget);
+
+    for (let i = 0; i < this.props.getCurrentViewTarget().tracks.length; i++) {
+      if (!tracks_equal(this.props.getCurrentViewTarget().tracks[i], this.getNextViewTarget().tracks[i])){
+        const viewTarget = this.getNextViewTarget();
```

Comment:
I don't think we want to call `getNextViewTarget()` so many times when we don't need to; it does some actual work if I remember right. `getCurrentViewTarget()` does less but still seems like work.
```suggestion

    const viewTarget = this.getNextViewTarget();
    const oldViewTarget = this.props.getCurrentViewTarget();
    for (let i = 0; i < oldViewTarget.tracks.length; i++) {
      if (!tracks_equal(oldViewTarget.tracks[i], viewTarget.tracks[i])){
```

Reply:
*Eventually* we're also going to need to handle the case of adding a track.